codegenerator/cli/npm/envio/src/db/EntityHistory.res

```diff
@@ -148,19 +160,13 @@ let insertRow = (
   self.insertFn(sql, row, ~shouldCopyCurrentEntity)
 }
 
-let batchInsertRows = (
-  self: t<'entity>,
-  ~sql,
-  ~rows: array<historyRow<'entity>>,
-  ~shouldCopyCurrentEntity,
-) => {
-  let rows =
-    rows
-    ->S.reverseConvertToJsonOrThrow(self.schemaRows)
-    ->(Utils.magic: Js.Json.t => array<Js.Json.t>)
+let batchInsertRows = (self: t<'entity>, ~sql, ~rows: array<historyRow<'entity>>) => {
   rows
-  ->Belt.Array.map(row => {
-    self.insertFn(sql, row, ~shouldCopyCurrentEntity)
```

Comment:
The bug creeps in because all entities in a rollback batch are treated as if they are part of the diff with the shouldCopyCurrentEntity flag.

This only makes a difference if

1. The entity update was a new one (not in the diff but maybe a new event emitted after the reorg)
2. Within the reorg threshold of the current reorg, a second reorg takes place, rolling back to the same block or before
3. The entity then appears in the diff as if it should be deleted (so to the framework looks like it doesn't exist in the db)
4. The handler has some conditional logic on the previous state of the entity (the conditional logic won't work as expected) OR the handler does not make any updates to the entity after the reorg in which case the entity would be unexpectedly deleted